From ca5ebcbea2cfa80c18c2ba73791c7bc4f0f306c8 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Tue, 9 Dec 2025 21:26:01 +0000
Subject: [PATCH 3/5] vendor: cilium/ebpf: use sha1cd for computing eBPF
 program tag

Use the vendored copy of sha1cd, the collision-detecting SHA1
implementation, to compute tags for eBPF programs. The tag is not
security-relevant, but its computation has to match the kernel
implementation (which uses SHA1 to compute the tag in kernel versions up
to and including Linux 6.17).

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 vendor/github.com/cilium/ebpf/asm/instruction.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vendor/github.com/cilium/ebpf/asm/instruction.go b/vendor/github.com/cilium/ebpf/asm/instruction.go
index 86b384c0..d641c1d9 100644
--- a/vendor/github.com/cilium/ebpf/asm/instruction.go
+++ b/vendor/github.com/cilium/ebpf/asm/instruction.go
@@ -1,7 +1,6 @@
 package asm
 
 import (
-	"crypto/sha1"
 	"encoding/binary"
 	"encoding/hex"
 	"errors"
@@ -12,6 +11,7 @@ import (
 	"strings"
 
 	"github.com/cilium/ebpf/internal/sys"
+	"github.com/pjbgf/sha1cd"
 )
 
 // InstructionSize is the size of a BPF instruction in bytes
@@ -793,7 +793,7 @@ func (insns Instructions) Marshal(w io.Writer, bo binary.ByteOrder) error {
 // to ProgramInfo.Tag to figure out whether a loaded program matches
 // certain instructions.
 func (insns Instructions) Tag(bo binary.ByteOrder) (string, error) {
-	h := sha1.New()
+	h := sha1cd.New()
 	for i, ins := range insns {
 		if ins.IsLoadFromMap() {
 			ins.Constant = 0
-- 
2.51.2

